indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -177,13 +178,23 @@ private boolean createAndStartSupervisorInternal(SupervisorSpec spec, boolean pe
       return false;
     }
 
-    Supervisor supervisor = spec.createSupervisor();
-    supervisor.start(); // try starting the supervisor first so we don't persist a bad spec
-
     if (persistSpec) {
       metadataSupervisorManager.insert(id, spec);
     }
 
+    Supervisor supervisor = null;
+    try {
+      supervisor = spec.createSupervisor();
+      supervisor.start();
+    }
+    catch (Exception e) {
+      // Supervisor creation or start failed write tombstone only when trying to start a new supervisor
```

Comment:
does supervisor.start() failure guarantee that things are not left in a corrupted state and supervisor.stop(true) does not need to be called?

@dclim ?